### DISCUSSION TITLE
Contact submission

### DISCUSSION THREAD

**Discussion by ciroiriarte** [JulianPrieber/littlelink-custom#619] [Feature requests]
Hello!,

It would be great if visitors could leave their contact information in a form:

email
name
phone number
maybe a link to their own LinkStack profile
maybe a note about where you met him/her

And when the user owner of the profile is logged in, to be able to download the contact as a vCard.

great for face 2 face events where you hand over a card with a QR code linking to your LinkStack.

Possible improvements for housekeeping:
- mark contacts already downloaded
- allow for manual deletion
- have a user defined automatic deletion/purge parameter (delete all the visitor/contacts entries that have already been downloaded more than X days ago)

**Comment by elsiehupp** [comment]
One thing that would be useful with this would be a sort of contact-info "handshake" with the following workflow:

* A visitor could enter their own contact information (e.g. email address or phone number) and an optional message in a web form;
* The LinkStack instance would send, e.g., a no-reply email to the email address entered for confirmation;
* Only once the visitor has confirmed that their, e.g., email address is valid would their message be delivered to the form owner.

I say "e.g. email" because it's the most obvious example: I'd kind of rather not have a bare email address out there for spambots to scrape. But companies like [Twilio](https://www.twilio.com/) offer SMS phone-number confirmation that be useful, as well, especially considering how much more sensitive phone numbers tend to be than email addresses.

The confirmation workflow for email could probably piggyback off the user-registration workflow, but a confirmation workflow for SMS would presumably require a paid API hook into something like Twilio. API access to, e.g., Twilio SMS could probably be folded into the cost of managed LinkStack hosting for people who don't feel like dealing with self-hosting, but it would make sense to facilitate the use of a custom API key for people who do choose to self-host.

(I assume there are competitors to Twilio, particularly outside the United States; I'm just familiar with Twilio in particular because I use their 2FA app [Twilio Authy](https://authy.com/).)

**Reply by MacBadger** [reply]
Having the ability to get this info would be supper helpful!

It could, at its basics, just be a customisation form, so people could use it for other things!
